engine/src/ast/function_expr.rs

```diff
@@ -164,8 +164,15 @@ impl<'i, 's> LexWith<'i, &'s Scheme> for FunctionCallExpr<'s> {
 
         let mut index = 0;
 
-        while let Some(',') = input.chars().next() {
-            input = skip_space(take(input, 1)?.1);
+        while let Some(c) = input.chars().next() {
+            if c == ')' {
```

Comment:
Overall, this parsing code should be simplified somehow, since it currently duplicates many checks for the same char, making flow harder to read. But it's ok to leave this to a separate PR.

Reply:
:+1:  I agree, i just did not wanted to refactor this piece in the same PR.